src/key.js

```diff
@@ -771,6 +771,84 @@ Key.prototype.verifyAllUsers = async function(keys) {
   return results;
 };
 
+/**
+ * generate a new subkey to a Key
+ * @param  {Object} options   (optional) options for creating subkey, e.g. {sign: true, passphrase: '123', algorithm:..., curve:...}
+ *
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+Key.prototype.generateSubkey = async function(options){
+  const defaultOptions = {};
+  if (this.primaryKey.algorithm.indexOf('rsa') === 0) {
+    defaultOptions.numBits = 2048;
+  }
+  else {
+    defaultOptions.curve = enums.curve.ed25519;
```

Comment:
Wouldn't it make more sense to default to the curve used by the primary key rather than always ed25519? And shouldn't the number of bits in the RSA case also default to the number of bits used by the primary key?

Reply:
I haven't found any methods or properties to get it directly yet.  I will read the code more carefully.  I know RSA well, But I have no idea about curve.

Reply:
I found the reason: the first byte of n(same security key) in V5 key is not the same in the V4 key.
So I can use the byteLength to check instead of BitLength.